typings/globalize/globalize.d.ts

```diff
@@ -1,5 +1,239 @@
+/// <reference path="../cldrjs/cldrjs.d.ts" />
+
 declare module 'globalize' {
+	import Cldr = require('cldr');
+
+	module Globalize {
+		export interface CurrencyFormatter {
+			(value: number): string;
+		}
+
+		export interface CurrencyOptions extends NumberOptions {
+			style?: string;
+		}
+
+		export interface DateFormatter {
+			(value: Date): string;
+		}
+
+		export interface DateParser {
+			(value: string): Date;
+		}
+
+		export interface DateOptions {
+			date?: string;
+			datetime?: string;
+			raw?: string;
+			skeleton?: string;
+			time?: string;
+		}
+
+		export interface MessageFormatter {
+			(variables: string | string[] | Object): string;
+			(...variables: string[]): string;
+		}
+
+		export interface NumberFormatter {
+			(value: number): string;
+		}
+
+		export interface NumberOptions {
+			maximumFractionDigits?: number;
+			maximumSignificantDigits?: number;
+			minimumFractionDigits?: number;
+			minimumIntegerDigits?: number;
+			minimumSignificantDigits?: number;
+			round?: string;
+			style?: string;
+			useGrouping?: Boolean;
+		}
+
+		export interface NumberParser {
+			(value: string): number;
+		}
+
+		export interface NumberParserOptions {
+			style?: string;
+		}
+
+		export interface Pluralizer {
+			(value: number): string;
+		}
+
+		export interface PluralOptions {
+			type: string;
+		}
+
+		export interface TimeFormatter {
+			(value: number): string;
+		}
+
+		export interface TimeOptions {
+			form: string;
+		}
+	}
+
 	class Globalize {
+		/**
+		 * @requires globalize/currency
+		 * @param currencyCode ISO 4217 (e.g. 'USD')
+		 */
+		static currencyFormatter(currencyCode: string, options?: Globalize.CurrencyOptions): Globalize.CurrencyFormatter;
```

Comment:
The 'globalize' module seems like it should follow the same layout as 'cldrjs', with separate declarations and exports for each submodule ('globalize/currency', etc.) to avoid the issue of typings not agreeing with loaded modules.